### ISSUE TITLE
Units not compatible with astropy.units used in parameter definition

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/simtools#537]
Requires some minor work to add `adc_counts` and `dimensionless` to be used as astropy.units. These are the only two types of units used in the parameter definitions which are not compatible with astropy.units.

Possible solution:

```
u.add_enabled_aliases({'adc_counts': u.ct })
u.add_enabled_aliases({'dimensionless': u.Unit('')})
```

**Comment by GernotMaier** [comment]
Do you mean "dimensionless" is strange or `u.Unit(''))? 

I don't have strong feelings about the first one, could change it to `dimensionless_unscaled` (although I don't think the unscaled is necessary).

The u.Unit is to only way I could find to tell astropy that this is dimensionless.

**Comment by orelgueta** [comment]
But it does that out of the box:

```
>>> from astropy import units as u
>>> u.Unit('')
Unit(dimensionless)
```

Or do you want astropy to read "dimensionless" and know it is equivalent to `""` (which is actually `dimensionless_unscaled`)

**Comment by GernotMaier** [comment]
Exactly.

**Comment by orelgueta** [comment]
OK, I think I understand then. Not sure we want to do this cause I assume there's a reason astropy decided to write `dimensionless_unscaled` rather than just `dimensionless`. And it doesn't seem like a big deal to simply write `dimensionless_unscaled ` everywhere. 
But maybe I am just being paranoid and it's harmless.

**Comment by GernotMaier** [comment]
No, the issue is (in my opinion) an inconsistency in astropy:

```
>>> import astropy.units as u
>>> a="km"
>>> u.Unit(a)
Unit("km")
>>> b="dimensionless_unscaled"
>>> u.Unit(b)
....
ValueError: 'dimensionless_unscaled' did not parse as unit: At col 0, dimensionless_unscaled is not a valid unit.  If this is meant to be a custom unit, define it with 'u.def_unit'. To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. For details, see https://docs.astropy.org/en/latest/units/combining_and_defining.html
>>> c=""
>>> u.Unit(c)
Unit(dimensionless)
```

This is very weird for several reasons:
- `dimensionless_unscaled` is not recognized (same for `dimensionless`
- only empty strings are recognized as dimensionless
- note that it prints `dimensionless` on the screen, not `dimensionless unscaled.

But:
```
>>> u.dimensionless_unscaled
Unit(dimensionless)
>>> u.dimensionless
AttributeError: module 'astropy.units' has no attribute 'dimensionless'.
```


So this is all over the place and why we need to define our custom unit for dimensionless.

**Comment by orelgueta** [comment]
Yes, I agree that this inconsistency is weird. I saw it as a result of this issue when I tried to figure out why you wanted to define `dimensionless`. 

When I made the first comment I assumed astropy had a good reason for this inconsistency. Now that I looked around a bit more I found that it does make sense in general (see [here](https://docs.astropy.org/en/stable/units/standard_units.html#the-dimensionless-unit) for the reason for `unscaled`). 

However, in our case it's probably not super important, so I don't really object to defining `dimensionless`.  If it does not lead to a lot of work or make the code less clear though, then I would vote for leaving the default astropy behaviour.
